admin/services/data-access/check-window.data.service.js

```diff
@@ -0,0 +1,84 @@
+'use strict'
+
+const moment = require('moment')
+
+const CheckWindow = require('../../models/check-window')
+
+const checkWindowDataService = {
+
+  /**
+   * Fetch check window document by id.
+   * @param id
+   * @returns {Promise.<*>}
+   */
+  fetchCheckWindow: async (id) => {
+    let checkWindow
+    checkWindow = await CheckWindow.findOne({'_id': id, 'deleted': false}).exec()
+    return checkWindow
+  },
+  /**
+   * Set check window as deleted.
+   * @param id
+   * @returns {Promise.<void>}
+   */
+  setDeletedCheckWindow: async (id) => {
+    let checkWindow
+    checkWindow = await CheckWindow.updateOne({'_id': id}, {$set: {'deleted': true}}).exec()
+    return checkWindow
+  },
+  /**
+   * Fetch check windows by status, sort by, sort direction and date (current or past).
+   * @param deleted
+   * @param sortBy
+   * @param sortDirection
+   * @param current
+   * @returns {Promise.<void>}
+   */
+  fetchCheckWindows: async (sortBy, sortDirection, deleted, current) => {
+    return new Promise(async (resolve, reject) => {
+      let checkWindows
+      let sort = {}
+      let query = {}
+
+      const currentTimestamp = moment.utc(Date.now()).format('YYYY-MM-DD HH:mm:ss.SSS')
+
+      sort[sortBy] = sortDirection
+      query.deleted = !deleted ? false : deleted
+      if (current === true) {
+        query.checkEndDate = {$gte: currentTimestamp}
+      } else {
+        query.checkEndDate = {$lt: currentTimestamp}
+      }
+
+      try {
+        checkWindows = await CheckWindow
+          .find(query)
+          .sort(sort)
+          .exec()
+      } catch (error) {
```

Comment:
Once you get rid of the promise you can just let this error escape - it will be converted to a reject() automatically